lib/AST/ModuleLoader.cpp

```diff
@@ -54,4 +59,73 @@ DependencyTracker::getClangCollector() {
   return clangCollector;
 }
 
+static bool findOverlayFilesInDirectory(SourceLoc diagLoc, StringRef path,
+                                        ModuleDecl *module,
+                                        DependencyTracker * const tracker) {
+  using namespace llvm::sys;
+  using namespace file_types;
+
+  ASTContext &ctx = module->getASTContext();
+  auto fs = ctx.SourceMgr.getFileSystem();
+
+  std::error_code error;
+  for (auto dir = fs->dir_begin(path, error);
+       !error && dir != llvm::vfs::directory_iterator();
+       dir.increment(error)) {
+    StringRef file = dir->path();
+    if (lookupTypeForExtension(path::extension(file)) != TY_SwiftOverlayFile)
+      continue;
+
+    module->addCrossImportOverlayFile(file);
+
+    // FIXME: Better to add it only if we load it.
+    if (tracker)
+      tracker->addDependency(file, module->isSystemModule());
+  }
+
+  if (error && error != std::errc::no_such_file_or_directory) {
+    ctx.Diags.diagnose(diagLoc, diag::cannot_list_swiftcrossimport_dir,
+                       module->getName(), error.message(), path);
+  }
+  return !error;
+}
+
+void ModuleLoader::findOverlayFiles(SourceLoc diagLoc, ModuleDecl *module,
+                                    FileUnit *file) {
+  using namespace llvm::sys;
+  using namespace file_types;
+
+  auto &langOpts = module->getASTContext().LangOpts;
+
+  SmallString<64> dirPath{file->getModuleDefiningPath()};
```

Comment:
I didn't realize you could use braces here instead of parens. Huh.

Reply:
That was added to C++ to deal with the [most vexing parse](https://en.wikipedia.org/wiki/Most_vexing_parse).